src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -360,12 +360,13 @@ class Crystal::Doc::Generator
     filename[@base_dir.size..-1]
   end
 
-  record RelativeLocation, filename : String, line_number : Int32, url : String? do
+  record RelativeLocation, filename : String, line_number : Int32, url : String?, duplicate : Bool do
```

Comment:
This logic should IMO be moved to the template.

Reply:
What do you mean by template? Should I check if it's a duplicate at the type.html ECR file?

Reply:
Yep, that's what I meant.

Reply:
Please don't. Templates should be mostly logic-less. This way is fine.

Reply:
It's not a `duplicate` though (still different line numbers).

Reply:
Should I rename it to `filename_duplicate` then? I wanted to keep the name simple and didn't tried to make the name very accurate.

Reply:
Don't use simple names. Use expressive names. As short as possible, though. I'd suggest something like `multiple_locations_in_file`.